test/CheckoutSdkTest/CheckoutSdkIntegrationTest.cs

```diff
@@ -39,6 +39,35 @@
                 ex.Message.ShouldBe("The API response status code (508) does not indicate success.");
             }
         }
+        
+        [Fact]
+        public async Task ShouldInstantiateClientWithSubdomainFactory()
+        {
+            var checkoutApi = CheckoutSdk
+                .Builder()
+                .Previous()
+                .StaticKeys()
+                .PublicKey(System.Environment.GetEnvironmentVariable("CHECKOUT_PREVIOUS_PUBLIC_KEY"))
+                .SecretKey(System.Environment.GetEnvironmentVariable("CHECKOUT_PREVIOUS_SECRET_KEY"))
+                .Environment(Environment.Sandbox)
+                .EnvironmentSubdomain(System.Environment.GetEnvironmentVariable("CHECKOUT_MERCHANT_SUBDOMAIN"))
+                .HttpClientFactory(new TestingClientFactory())
+                .Build();
+
+            checkoutApi.ShouldNotBeNull();
+
+            try
+            {
+                await checkoutApi.EventsClient().RetrieveAllEventTypes();
+                throw new XunitException();
+            }
+            catch (Exception ex)
+            {
+                ex.ShouldNotBeNull();
+                ex.ShouldBeAssignableTo(typeof(CheckoutApiException));
+                ex.Message.ShouldBe("The API response status code (508) does not indicate success.");
+            }
```

Comment:
Can you refactor this to avoid this check?  You could have a specific catch clause for CheckoutApiException and then a general exception (I'd hope CodeQL would be ok with that, but 🤷 )